app/main/views/frameworks.py

```diff
@@ -750,10 +750,20 @@ def framework_agreement(framework_slug):
 
     # if there's a frameworkAgreementVersion key, it means we're on G-Cloud 8 or higher
     if framework.get('frameworkAgreementVersion'):
+        def lot_result(drafts_for_lot):
+            if any(draft['status'] == 'submitted' for draft in drafts_for_lot):
+                return 'Successful'
+            elif any(draft['status'] == 'failed' for draft in drafts_for_lot):
+                return 'Unsuccessful'
+            else:
+                return 'No application'
```

Comment:
That's true.  I can't see a particularly neat way round it though, so going to suck it up.